api/models/Tag.js

```diff
@@ -15,7 +15,7 @@ var addToTaggable = (taggable, tagName, selected, trx) => {
     association = 'post.communities'
     communities = comment => comment.relations.post.relations.communities.models
   }
-  return taggable.load(association)
+  return taggable.load(association, {transacting: trx})
   .then(() => Tag.find(tagName))
   .then(tag => {
     if (tag) {
```

Comment:
"catch the save error and find again" works in general -- i'm not totally sure how it behaves in this transaction context though, i.e. does the error surface immediately or only when you try to commit the transaction.